RELEASE/scripts/autoscend.ash

```diff
@@ -2709,24 +2711,49 @@ boolean LX_getDigitalKey()
 	{
 		return false;
 	}
+	if(item_amount($item[Digital Key]) > 0)
+	{
+		if(have_effect($effect[Consumed By Fear]) > 0)
+		{
+			uneffect($effect[Consumed By Fear]);
+			council();
+		}
+		return false;
+	}
+	
+	//why is this here?
 	if (internalQuestStatus("questL12War") == 1 && get_property("sidequestArenaCompleted") != "none" && get_property("auto_powerLevelAdvCount").to_int() < 9)
 	{
 		return false;
 	}
-	while((item_amount($item[Red Pixel]) > 0) && (item_amount($item[Blue Pixel]) > 0) && (item_amount($item[Green Pixel]) > 0) && (item_amount($item[White Pixel]) < 30) && (item_amount($item[Digital Key]) == 0))
+	
+	//craft the key if you can
+	while(creatable_amount($item[white pixel]) > 0 && (item_amount($item[White Pixel]) < 30) && item_amount($item[Digital Key]) == 0)
 	{
-		cli_execute("make white pixel");
+		//is making white pixels individually even necessary? test and if not remove this function entirely.
+		create(1, $item[White Pixel]);
 	}
-	if((item_amount($item[white pixel]) >= 30) || (item_amount($item[Digital Key]) > 0))
+	if(creatable_amount($item[Digital Key]) > 0 && item_amount($item[Digital Key]) == 0)
 	{
-		if(have_effect($effect[Consumed By Fear]) > 0)
+		if(create(1, $item[Digital Key]))
 		{
-			uneffect($effect[Consumed By Fear]);
-			council();
+			return true;
+		}
+		else
+		{
+			abort("Mysteriously failed to craft [Digital Key] even though I should be able to make one. Make it manually then run me again");
 		}
+	}
+	
+	//Powerful Glove is an IOTM accessory that drops lots of white pixels. If you have it you do not want to waste adv farming digital key.
+	//You still want to craft digital key though, which is why this ie below the crafting section.
+	if(auto_hasPowerfulGlove())
```

Comment:
What if we make it to the tower, have no other quests available, and don't have enough white pixels to craft the digital key?

Reply:
commit ee9e3df